code/modules/urist/modules/jungle/planet_turfs.dm

```diff
@@ -11,96 +11,117 @@
-		if(!bushes)
-			new small_tree_type(src)
+/turf/simulated/floor/planet/ex_act(severity)
+	return
 
-	if(large_trees_chance && prob(large_trees_chance ))
-		if(!bushes)
-			new large_tree_type(src)
+/turf/simulated/floor/planet/on_update_icon(update_neighbors = FALSE)
+	. = ..()
 
-	if(animal_spawn_chance && prob(animal_spawn_chance ))
-		if(prob(50)) //even 1% is a fuck ton. A 100x100 area is 10000 tiles. This means that at 1% chance, 100 animals will spawn in that area.
-			var/A = pick(animal_spawn_list) //A conservative estimate for possible spawning tiles in the jungle would be 22500, which is very conservative. This means that roughly 225 animals would spawn.
-			new A(get_turf(src)) //thus, we half that number, which leads to more sane numbers, and I don't have to make every spawn a fraction of 1.
+	if(plant_overlay)
+		plant_overlay = FALSE
+		var/image/I
+		if(prob(35))
+			I = image('icons/jungle.dmi',"plant[rand(1,7)]")
+		else
+			if(prob(30))
+				I = image('icons/obj/flora/ausflora.dmi',"reedbush_[rand(1,4)]")
+			else if(prob(33))
+				I = image('icons/obj/flora/ausflora.dmi',"leafybush_[rand(1,3)]")
+			else if(prob(50))
+				I = image('icons/obj/flora/ausflora.dmi',"fernybush_[rand(1,3)]")
+			else
+				I = image('icons/obj/flora/ausflora.dmi',"stalkybush_[rand(1,3)]")
+		I.pixel_x = rand(-6,6)
+		I.pixel_y = rand(-6,6)
+		AddOverlays(I)
 
-	if(trap_spawn_chance)
-		if(!bushes && prob(0.5))
-			new /obj/structure/pit/punji6/hidden/dull(src)
+/turf/simulated/floor/planet/proc/generate_planet_objects()
+	var/bushes = FALSE
+	var/tree = FALSE
+
+	if(plants_spawn_chance)
+		if(prob(plants_spawn_chance)) //why were we running prob() on all of these even if they don't fucking spawn there
+			if(prob(90))
+				plant_overlay = TRUE
+				queue_icon_update()
+			else
+				spawn_list.Add(/obj/structure/jungle_plant)
+
+	if(misc_plant_spawn_chance)
+		if(prob(misc_plant_spawn_chance))
+			spawn_list.Add(misc_plant_type)
+
+	if(bushspawnchance)
+		if(prob(bushspawnchance))
+			spawn_list.Add(bush_type)
+			bushes = TRUE
+
+	if(!bushes)
+		if(small_trees_chance)
+			if(prob(small_trees_chance)) //one in four give or take, we'll see how that goes. //IT WENT TERRIBLY
+				spawn_list.Add(small_tree_type)
+				tree = TRUE
+		if(!tree && large_trees_chance)
+			if(prob(large_trees_chance))
+				spawn_list.Add(large_tree_type)
+		if(trap_spawn_chance)
+			if(prob(0.1))
+				spawn_list.Add(/obj/structure/pit/punji6/hidden/dull)
+
+	if(length(animal_spawn_list))
+		if(prob(animal_spawn_chance)) //even 1% is a fuck ton. A 100x100 area is 10000 tiles. This means that at 1% chance, 100 animals will spawn in that area.
+			var/A = pick(animal_spawn_list) //A conservative estimate for possible spawning tiles in the jungle would be 22500, which is very conservative. This means that roughly 225 animals would spawn.
+			spawn_list.Add(A) //so set your numbers fucking low. god i hate this shit.
```

Comment:
Rather than statically spawning them per-tile, might be nicer to use the uspawner objects I've added a while back.

You can set the spawn radius on them or set them to target the whole area, and you can set the max spawn limit and adjust spawn tickrates if desired. If using endless, this would emulate animals migrating in.

There's currently no tracking for how many of these are living so that we could spawn based on active mobs, but some Observer magic would almost certainly easily handle that now that I know it exists.